### PR TITLE
Publish image for infinispan-test, replaces Jenkins stage 'Image'

### PR DESCRIPTION
**Pull request by rigazilla**
Workflow to build and publish infinispan server image on quay.io/infinispan-test

### CODE REVIEW

File: .github/workflows/on_call_do_publish_image.yml [hunk 1]
```diff
@@ -0,0 +1,115 @@
+name: CI Release Images
+
+on:
+  workflow_call:
+    inputs:
+      repository:
+        description: 'Name of the repo to download server artifact from'
+        type: string
+      run-id:
```
Comment (ryanemerson) [inline .github/workflows/on_call_do_publish_image.yml:9]: Do we need to explicitly specify this? From the GH page:

```
    # The id of the workflow run where the desired download artifact was uploaded from.
    # If github-token is specified, this is the run that artifacts will be downloaded from.
    # Optional. Default is ${{ github.run_id }}
```

File: .github/workflows/on_call_do_publish_image.yml [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        description: 'Id of the run to download server artifact from'
+        type: string
+      art-name:
```
Comment (ryanemerson) [inline .github/workflows/on_call_do_publish_image.yml:12]: IMO `artifact` or `serverArtifact` would be more descriptive.

File: .github/workflows/on_call_do_publish_image.yml [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+         description: 'Artifact name to download'
+         type: string
+      tags:
+        description: 'Tags for the image'
+        type: string
+      push:
+         description: 'If true push image.'
+         default: true
+         type: boolean
+
```
Comment (ryanemerson) [inline .github/workflows/on_call_do_publish_image.yml:22]: Redundant space

File: .github/workflows/on_call_do_publish_image.yml [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+         type: string
+    secrets:
+      token:
+         required: true
+         description: 'GH token'
+      quay_user:
+         required: true
+      quay_pass:
+         required: true
+  workflow_dispatch:
```
Comment (ryanemerson) [inline .github/workflows/on_call_do_publish_image.yml:35]: Maybe add a new line here to separate the two call types?

File: .github/workflows/on_call_do_publish_image.yml [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+           run: |
+               pwd
+               ls
+               SERVER_OVERRIDE="{\"artifacts\":[{\"name\":\"server\",\"path\":\"${GITHUB_WORKSPACE}/${SERVER_ZIP}\"}]}"
+               TAG="${TAGS%%,*}"
+               IMG_NAME="${TAG%%:*}"
+               IMG_VER="${TAG##*:}"
+               cekit -v --descriptor server-openjdk.yaml build --overrides '{"version": "'${IMG_VER}'", "name": "'${IMG_NAME}'"}' --overrides ${SERVER_OVERRIDE} --dry-run docker
+
+         - name: Set up Docker Buildx
```
Comment (ryanemerson) [inline .github/workflows/on_call_do_publish_image.yml:96]: I think the flow of the action is clearer as a reader if we move all of the dependency and initialization steps to the start of the job, followed by the actual logic of the action. A benefit of this, is that if any of the required dependencies/login steps fail, we're failing-fast and no steps are executed redundantly.

**Review by ryanemerson — COMMENTED** [review]
Thanks @rigazilla, overall LGTM just a few minor things. Do you have a link to the corresponding Infinispan action? If I can see the user of this, it will give me the full context to review more thoroughly.

**Comment by ryanemerson** [comment]
Very minor nitpick, I would rename the action to just `publish_image.yml` as we have both call and dispatch targets.

**Comment by rigazilla** [comment]
> Thanks @rigazilla, overall LGTM just a few minor things. Do you have a link to the corresponding Infinispan action? If I can see the user of this, it will give me the full context to review more thoroughly.

It will be based on this: https://github.com/rigazilla/infinispan/blob/main/.github/workflows/on_run_publish_image.yml
it needs some updates first

File: .github/workflows/on_call_do_publish_image.yml (hunk 1 above)
Comment (rigazilla) [reply .github/workflows/on_call_do_publish_image.yml:9]: no, it's not called directly from the pr/push workflow, due to security reasons. so the initial run id must be passed as input

**Comment by rigazilla** [comment]
> Very minor nitpick, I would rename the action to just `publish_image.yml` as we have both call and dispatch targets.

yeah ok, the naming convention I'm trying to apply doesn't work very well... too verbose

**Comment by ryanemerson** [comment]
> It will be based on this: https://github.com/rigazilla/infinispan/blob/main/.github/workflows/on_run_publish_image.yml
> it needs some updates first

Thanks, that's really helpful.

So looking at you current code, it seems like for PRs we need to wait for all tests to run before the image is built. This makes sense for pushes to branches, but would it be possible to make it so that the workflow is triggered on PR open? Ideally I wouldn't have to wait ~1h before the image build happens.

File: .github/workflows/on_call_do_publish_image.yml (hunk 1 above)
Comment (ryanemerson) [reply .github/workflows/on_call_do_publish_image.yml]: Argh makes sense, I didn't realise we weren't calling it direct.

**Comment by rigazilla** [comment]
> > It will be based on this: https://github.com/rigazilla/infinispan/blob/main/.github/workflows/on_run_publish_image.yml
> > it needs some updates first
> 
> Thanks, that's really helpful.
> 
> So looking at you current code, it seems like for PRs we need to wait for all tests to run before the image is built. This makes sense for pushes to branches, but would it be possible to make it so that the workflow is triggered on PR open? Ideally I wouldn't have to wait ~1h before the image build happens.

oh that was my intent, but is gone lost when I moved out the publishing code from the main workflow. I'll work on that

if this PR looks good, is it ok for a merge? 
this way I can point the main branch instead of this in the other workflows and simplify things

File: .github/workflows/publish_image.yml [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+  workflow_dispatch:
+    inputs:
+      repository:
+        description: 'Name of the repo to download server artifact from'
+        type: string
+      run-id:
+        description: 'Id of the run to download server artifact from'
+        type: string
+      serverArtifact:
```
Comment (ryanemerson) [inline .github/workflows/publish_image.yml:43]: I don't mind whether we use camelCase or kebab-case, but let's try to be consistent with the input element across our various workflows.

**Comment by ryanemerson** [comment]
> if this PR looks good, is it ok for a merge?

Do you have a link to a successful run from your fork?

**Comment by rigazilla** [comment]
> > if this PR looks good, is it ok for a merge?
> 
> Do you have a link to a successful run from your fork?

yep, I'm running on my fork for now:
- [publish for PR](https://github.com/rigazilla/infinispan/actions/runs/10833921459)
- [publish for PUSH](https://github.com/rigazilla/infinispan/actions/runs/10833490530)

[published images](https://quay.io/repository/rigazilla/server-test?tab=tags)

File: .github/workflows/publish_image.yml (hunk 6 above)
Comment (rigazilla) [reply .github/workflows/publish_image.yml:43]: let's go with camel

**Comment by ryanemerson** [comment]
Thanks @rigazilla

**Pull request merged**